insights/client/connection.py

```diff
@@ -692,7 +692,10 @@ def upload_archive(self, data_collected, duration, cluster=None):
 
         logger.debug("Upload status: %s %s %s",
                      upload.status_code, upload.reason, upload.text)
-        the_json = json.loads(upload.text)
+        if upload.status_code == 201:
+            the_json = json.loads(upload.text)
+        else:
+            logger.debug("Upload archive failed with status code  %s" % upload.status)
```

Comment:
Should we bail out here?  If `the_json` isn't set then the rest of the function is in a bad state.